base/cvd/cuttlefish/host/commands/cvd/server_command/status_fetcher.cpp

```diff
@@ -224,10 +224,28 @@ Result<StatusFetcherOutput> StatusFetcher::FetchStatus(
 }
 
 Result<Json::Value> StatusFetcher::FetchGroupStatus(
-    const selector::LocalInstanceGroup& group,
+    selector::LocalInstanceGroup& group,
     const RequestWithStdio& original_request) {
   Json::Value group_json(Json::objectValue);
   group_json["group_name"] = group.GroupName();
+  std::set<cvd::InstanceState> reachable_states{
+      cvd::InstanceState::RUNNING, cvd::InstanceState::STARTING,
+      cvd::InstanceState::UNREACHABLE};
+  CF_EXPECT(!group.Instances().empty(), "Group can't be empty");
+  if (!Contains(reachable_states, group.Instances()[0].State())) {
```

Comment:
Should this be checked for every instance rather than `Instances()[0]`?

Reply:
Done

I was under the (wrong) impression that only one control socket was contacted for the entire group. Ended up rewriting most of my changes in this file and adding a `UpdateInstance` function to isntance manager and database.